src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+			for (Entry<String, String> requirements : orderRequirements.entrySet()) {
+				resultList = filterFlavors(requirements);
+				if (resultList.size() < resultSet.size()) {
+					resultSet = parseToTreeSet(resultList);
+				}
+			}
+		}
+		return resultSet;
+	}
+	
+	protected TreeSet<AwsHardwareRequirements> parseToTreeSet(List<AwsHardwareRequirements> list) {
+		TreeSet<AwsHardwareRequirements> resultSet = new TreeSet<AwsHardwareRequirements>();
+		resultSet.addAll(list);
+		return resultSet;
+	}
+
+	protected List<AwsHardwareRequirements> filterFlavors(Entry<String, String> requirements) {
+		String key = requirements.getKey().trim();
+		String value = requirements.getValue().trim();
+		return getFlavors().stream()
+				.filter(flavor -> flavor.getRequirements().get(key).equalsIgnoreCase(value))
+				.collect(Collectors.toList());
+	}
+	
+	protected TreeSet<AwsHardwareRequirements> getFlavors() {
+		synchronized (this.flavors) {
+			return this.flavors;
+		}
+	}
+
+	protected void updateHardwareRequirements(AwsV2User cloudUser)
+			throws ConfigurationErrorException, InvalidParameterException, UnexpectedException {
+
+		List<String> lines = loadLinesFromFlavorFile();
+		String[] requirements = null;
+		AwsHardwareRequirements flavor = null;
+
+		Map<String, Integer> imagesMap = getImagesMap(cloudUser);
+		for (Entry<String, Integer> images : imagesMap.entrySet()) {
+			for (String line : lines) {
+				if (!line.startsWith(COMMENTED_LINE_PREFIX)) {
+					requirements = line.split(CSV_COLUMN_SEPARATOR);
+					flavor = mountHardwareRequirements(images, requirements);
+					this.flavors.add(flavor);
+				}
+			}
+		}
+	}
+
+	protected AwsHardwareRequirements mountHardwareRequirements(Entry<String, Integer> image, String[] requirements) {
+		String name = requirements[INSTANCE_TYPE_COLUMN];
+		String flavorId = getRandomUUID();
+		int cpu = Integer.parseInt(requirements[VCPU_COLUMN]);
+		Double memory = Double.parseDouble(requirements[MEMORY_COLUMN]) * ONE_GIGABYTE;
+		int disk = image.getValue();
+		String imageId = image.getKey();
+		Map<String, String> requirementsMap = loadRequirementsMap(requirements);
+		return new AwsHardwareRequirements(name, flavorId, cpu, memory.intValue(), disk, imageId, requirementsMap);
+	}
+
+	protected Map<String, String> loadRequirementsMap(String[] requirements) {
+		Map<String, String> requirementsMap = new HashMap<String, String>();
+		String processor = requirements[PROCESSOR_COLUMN];
+		String performance = requirements[NETWORK_PERFORMANCE_COLUMN];
+		String storage = requirements[STORAGE_COLUMN];
+		String bandwidth = requirements[DEDICATED_EBS_BANDWIDTH_COLUMN];
+		requirementsMap.put(PROCESSOR_REQUIREMENT, processor);
+		requirementsMap.put(PERFORMANCE_REQUIREMENT, performance);
+		requirementsMap.put(STORAGE_REQUIREMENT, storage);
+		requirementsMap.put(BANDWIDTH_REQUIREMENT, bandwidth);
+		return requirementsMap;
+	}
+
+	protected Map<String, Integer> getImagesMap(AwsV2User cloudUser)
```

Comment:
There is a very similar method in imagePlugin, we can use it?

Reply:
Unfortunately, I know there is no way to instantiate a plugin inside other.